SU2_CFD/include/drivers/CDiscAdjMultizoneDriver.hpp

```diff
@@ -0,0 +1,183 @@
+ *  - Prof. Nicolas R. Gauger's group at Kaiserslautern University of Technology.
+ *  - Prof. Alberto Guardone's group at Polytechnic University of Milan.
+ *  - Prof. Rafael Palacios' group at Imperial College London.
+ *  - Prof. Vincent Terrapon's group at the University of Liege.
+ *  - Prof. Edwin van der Weide's group at the University of Twente.
+ *  - Lab. of New Concepts in Aeronautics at Tech. Institute of Aeronautics.
+ *
+ * Copyright 2012-2019, Francisco D. Palacios, Thomas D. Economon,
+ *                      Tim Albring, and the SU2 contributors.
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+#include "CMultizoneDriver.hpp"
+
+class CDiscAdjMultizoneDriver : public CMultizoneDriver {
+
+protected:
+
+  /*!
+   * \brief Kinds of recordings (three different ones).
+   */
+  enum ENUM_TAPE {
+    FULL_TAPE = 1,                /*!< \brief Entire derivative information for a coupled adjoint
+                                              solution update. */
+    OBJECTIVE_FUNCTION_TAPE = 2,  /*!< \brief Record only the dependence of the objective function
+                                              w.r.t. solver variables (from all zones). */
+    ZONE_SPECIFIC_TAPE = 3        /*!< \brief Record only the dependence of the solution update in a
+                                              specified zone w.r.t. solver variables (from all zones). */
+  };
+
+  /*!
+   * \brief Position markers within a tape.
+   */
+  enum ENUM_TAPE_POSITIONS {
+    START = 0,                    /*!< \brief Beginning of the tape. */
+    REGISTERED = 1,               /*!< \brief Solver variables are registered on the tape. */
+    DEPENDENCIES = 2,             /*!< \brief Derived values (e.g. gradients) are set. */
+    OBJECTIVE_FUNCTION = 3,       /*!< \brief Objective function is set. */
+    TRANSFER = 4,                 /*!< \brief Solution data is transferred between coupled solvers of
+                                              different physical zones (giving cross contributions). */
+    ITERATION_READY = 5           /*!< \brief Until here, all derivative information is gathered so
+                                              that it can be connected to a solver update evaluation. */
+  };
+
+  unsigned short RecordingState;  /*!< \brief The kind of recording that the tape currently holds. */
```

Comment:
can you use the right enum type instead of unsigned short here? I think it improves readability.

Reply:
Sure :)